infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/cache/jpa/JpaKubernetesRuntimeStateCache.java

```diff
@@ -211,6 +233,21 @@ protected void doUpdateStatus(
     entityManager.flush();
   }
 
+  @Transactional(rollbackOn = {RuntimeException.class, InfrastructureException.class})
+  protected void doUpdateCommands(RuntimeIdentity id, List<KubernetesRuntimeCommandImpl> commands)
+      throws InfrastructureException {
+    Optional<KubernetesRuntimeState> runtimeStateOpt = get(id);
+
+    if (!runtimeStateOpt.isPresent()) {
+      throw new InfrastructureException(
+          "Runtime state for workspace with id '" + id.getWorkspaceId() + "' was not found");
+    }
+
+    runtimeStateOpt.get().setCommands(commands);
+
+    managerProvider.get().flush();
```

Comment:
Isn't `get()` should be before actual saving (i see it is done this way in most places)

Reply:
I think it doesn't really matter.